src/sweeps/bayes_search.py

```diff
@@ -324,12 +331,60 @@ def next_sample(
     )
 
 
+def impute(
+    goal: str,
+    metric_name: str,
+    impute_strategy: ImputeStrategy,
+    run: Optional[SweepRun] = None,
```

Comment:
why do we accept both `run` and `runs`? you seem to shadow the `run` variable in places, can we just accept `runs`?

Reply:
worst needs all the runs in the sweep to calculate the worst metric across all runs, the other two methods only need the target run to calculate the best metric or latest metric from that run.